### PR TITLE
Add distribution version to the ProjectAPI usage

### LINKED ISSUE #30641: Project API throws Transitive Dependency Cannot be Found
**Description:**
In a Java Micro Service which uses Ballerina Project API to parse a Ballerina build project we get a Transitive Dependency Error for three connectors. The three connectors got pulled were older versions and were incompatible with SL Alpha 5.3 version of Ballerina which was used in our Java Micro Service. The version numbers which got pulled are also mentioned in the below list.

- Azure Event Hub 0.1.1 (Transitive dependency cannot be found: org=ballerina, package=file, version=0.7.0-alpha4)
- Mongo DB 2.0.4 (Transitive dependency cannot be found: org=ballerina, package=time, version=1.1.0-alpha4])
- AWS S3 0.99.2 (Transitive dependency cannot be found: org=ballerina, package=file, version=0.7.0-alpha4)

However, we have been informed by the connector team that all the above three connectors had the latest version upgraded to Ballerina Sl Alpha 5 and those version were also available in Ballerina central by the time the transitive dependency error occurred. The list of latest versions available in the Ballerina Central were as follows.

- Azure Event Hub - 0.1.3
- Mongo DB - 2.0.7
- AWS S3 - 0.99.4

The top part of the error log which got printed by the Java Micro service is as follows,

```
io.ballerina.projects.ProjectException: Transitive dependency cannot be found: org=ballerina, package=file, version=0.7.0-alpha4
        at io.ballerina.projects.internal.PackageDependencyGraphBuilder.buildPackageDependencyGraph(PackageDependencyGraphBuilder.java:203) ~[ballerina-lang-2.0.0-alpha8-20210513-185400-180f6504.jar:na]
        at io.ballerina.projects.PackageResolution.buildDependencyGraph(PackageResolution.java:153) ~[ballerina-lang-2.0.0-alpha8-20210513-185400-180f6504.jar:na]
        at io.ballerina.projects.PackageResolution.<init>(PackageResolution.java:71) ~[ballerina-lang-2.0.0-alpha8-20210513-185400-180f6504.jar:na]
        at io.ballerina.projects.PackageResolution.from(PackageResolution.java:78) ~[ballerina-lang-2.0.0-alpha8-20210513-185400-180f6504.jar:na]
        at io.ballerina.projects.PackageContext.getResolution(PackageContext.java:213) ~[ballerina-lang-2.0.0-alpha8-20210513-185400-180f6504.jar:na]
        at io.ballerina.projects.PackageCompilation.<init>(PackageCompilation.java:68) ~[ballerina-lang-2.0.0-alpha8-20210513-185400-180f6504.jar:na]
        at io.ballerina.projects.PackageCompilation.from(PackageCompilation.java:94) ~[ballerina-lang-2.0.0-alpha8-20210513-185400-180f6504.jar:na]
        at io.ballerina.projects.PackageContext.getPackageCompilation(PackageContext.java:206) ~[ballerina-lang-2.0.0-alpha8-20210513-185400-180f6504.jar:na]
        at io.ballerina.projects.Package.getCompilation(Package.java:131) ~[ballerina-lang-2.0.0-alpha8-20210513-185400-180f6504.jar:na]
        
```
Since the latest versions of the connector were available in the Ballerina central and we did not have any dependency locking in `Dependencies.toml` file of the build project I believe the latest version of the above mentioned three connectors should have been pulled from the Ballerina Central.

The workaround we used to fix the above error was to explicitly mention the three connector versions in the `Dependencies.toml` file of the build project as a temporary fix.

**Steps to reproduce:**
If one tried to load a Ballerina build project using project API and invoke project API's `package.getCompilation()` method using the below mentioned ballerina version without any connector versions mentioned in the `Dependencies.toml` file this issue could be reproduced.

**Affected Versions:**
Java Micro Service was built using Java 11, Ballerina SL Alpha 5.3 with Ballerina Lang version `2.0.0-alpha8-20210513-185400-180f6504`

**OS, DB, other environment details and versions:**
Ubuntu 20.04.2.0

### PR DESCRIPTION
**Pull request by IrushiL**
## Purpose
$subject

Fixes #30641

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by azinneera — COMMENTED** [review]
Can't we move it from the cli module instead of duplicating the resource file in ballerina-lang module

**Comment by IrushiL** [comment]
> Can't we move it from the cli module instead of duplicating the resource file in ballerina-lang module

This is used in the `Main` CLI launcher class for printing the version information, therefore, I can't see a way how we could completely move this file.

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java [hunk 1]
```diff
@@ -734,7 +734,7 @@ public static String getBallerinaPackVersion() {
      * @return ballerina short version
      */
     public static String getBallerinaShortVersion() {
-        try (InputStream inputStream = ProjectUtils.class.getResourceAsStream(ProjectDirConstants.PROPERTIES_FILE)) {
+        try (InputStream inputStream = ProjectUtils.class.getResourceAsStream(PROPERTIES_FILE)) {
             var properties = new Properties();
             properties.load(inputStream);
             return properties.getProperty(ProjectDirConstants.BALLERINA_SHORT_VERSION);
```
Comment (IrushiL) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java:740]: Noticed that both `BALLERINA_SHORT_VERSION` and `BALLERINA_VERSION` constants point to the same property. Is this intentional?

**Comment by azinneera** [comment]
> > Can't we move it from the cli module instead of duplicating the resource file in ballerina-lang module
> 
> This is used in the `Main` CLI launcher class for printing the version information, therefore, I can't see a way how we could completely move this file. However, I removed the unused fields from the `tool.properties` file.

ballerina-lang is a dependency of the cli module, therefore, it will be added to the execution classpath of the cli launcher. So we don't need to have it in both modules

**Review by azinneera — APPROVED** [review]
LGTM

**Pull request merged**